www/js/render-solution.js

```diff
@@ -93,6 +102,24 @@ define([
 				this.showMessage(sol, "graphErrorMessage", "error", false);
 			}
 			this.resizeWindow();
+			var helpButton = registry.byId("graphHelpButton");
+
+			on(helpButton, "click", lang.hitch(this, function () {
+				if(!this.helpDialog) {
+					var content = "<ul>";
+					for(var msg in this._helpText){
+						content += "<li>" + this._helpText[msg] + "</li>";
+					}
+					content += "</ul>";
+					this.helpDialog = new Dialog({
+						"class": 'graphHintUnderLay',
+						"title": "Graph Help",
+						content: content
+					});
```

Comment:
can we make the code to less intrusive. What I mean to say is that you have help texts everywhere in all the files. I just saw one in pedagogical_module. now this is in render-solution.js. If possible please move these codes out. U only need to put an event on helpButton which u can do here if you want and then use that to call this piece of code where all the code for messages is kept in one file. Something I tried for logging messages.